File: EducNotes.API/Models/Lesson.cs
namespace EducNotes.API.Models
{
    public class Lesson
    {
        public int Id { get; set; }
        public int? ClassLevelId { get; set; }
        public ClassLevel ClassLevel { get; set; }
        public int? CourseId { get; set; }
        public Course Course { get; set; }
        public int? ThemeId { get; set; }
        public Theme Theme { get; set; }
        public string Name { get; set; }
        public string Desc { get; set; }
    }
}

File: EducNotes-SPA/src/app/_models/lesson.ts
import { ClassLevel } from './classLevel';
import { Theme } from './theme';
import { Course } from './course';
import { LessonContent } from './lesson-content';

// import {} from 'class-validator';

export class Lesson {
  public id: number;

  public classLevelId: number;

  public classLevel: ClassLevel;

  public courseId: number;

  public course: Course;

  public themeId: number;

  public theme: Theme;

  public name: string;

  public desc: string;
  public lessonContents: LessonContent[];

  constructor(
    id?: number,
    classLevelId?: number,
    classLevel?: ClassLevel,
    courseId?: number,
    course?: Course,
    themeId?: number,
    theme?: Theme,
    name?: string,
    desc?: string,
  ) {
    this.id = id || 0;
    this.classLevelId = classLevelId || null;
    this.courseId = courseId || null;
    this.themeId = themeId || null;
    this.name = name || '';
    this.desc = desc || '';
  }
}
